site/_utils/addCollectionByDirectory.js

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file Creates a collection for a directory under a specific locale
+ * while using the same path in the default locale as a fallback.
+ * For example used to show localized (or original) blog articles
+ * alongside untranslated ones.
+ */
+
+const path = require('path');
+const {filterOutDrafts} = require('./drafts');
+
+const defaultLocale = 'en';
+
+function addCollectionByDirectory(config, locale, dir) {
+  config.addCollection(`${dir}-${locale}`, collections => {
+    const baseCollection = collections
+      .getFilteredByGlob(
+        path.join('.', 'site', defaultLocale, dir, '*', '*.md')
+      )
+      .filter(filterOutDrafts)
+      .reverse();
+
+    // Test case: if we're running inside of Percy then don't care
+    // about translations. Just show the first six posts.
+    if (process.env.PERCY_BRANCH) {
+      return baseCollection.slice(baseCollection.length - 6);
+    }
+
+    // If we are building the collection for the default locale,
+    // there also is no need to compare with English itself
+    if (locale === defaultLocale) {
+      return baseCollection;
+    }
+
+    const localeCollection = collections
+      .getFilteredByGlob(path.join('.', 'site', locale, dir, '*', '*.md'))
+      .filter(filterOutDrafts)
+      .reverse();
+
+    // If there aren't any translations for that directory there is no
+    // sense in trying to merge them, just return the English articles then
+    if (localeCollection.length === 0) {
+      return baseCollection;
+    }
+
+    // Otherwise walk through the English collection while looking
+    // for matches in the locale collection and if it's a match, replace
+    let localeIndex = 0;
+    let collection = [];
+    for (let index = 0; index < baseCollection.length; index++) {
+      const item = baseCollection[index];
+      const localizedItem = localeCollection[localeIndex];
+
+      if (item.fileSlug === localizedItem.fileSlug) {
+        collection.push(localeCollection[localeIndex]);
+        localeIndex++;
+      } else {
+        collection.push(baseCollection[index]);
+      }
+
+      // If there are no items left to potentially localize then simply
+      // append the rest of the articles and stop
+      if (localeIndex === localeCollection.length) {
+        collection = collection.concat(baseCollection.slice(index + 1));
```

Comment:
11ty collections are sorted by default (by date *I think*), you'll loose this behavior here. There might be parts of the code that silently rely on this characteristic, so maybe worth seorting the collection here.

Reply:
Clarified offline. The code works as intended under the assumption that `localeCollection` has the same order as `baseCollection` and is a translated subset of it, not containing any original items.

Agreed to only rely on slugs for matching instead of on date order as well.